eng/release/DiagnosticsReleaseTool/Core/Release.cs

```diff
@@ -240,12 +240,22 @@ private async Task<int> LayoutFilesAsync(CancellationToken ct)
                             (FileMapping fileMap, FileMetadata fileMetadata) = layoutResultArray[i];
                             string srcPath = fileMap.LocalSourcePath;
                             string dstPath = fileMap.RelativeOutputPath;
-                            if (relativePublishPathsUsed.Contains(dstPath))
+                            if (relativePublishPathsToHash.ContainsKey(dstPath))
                             {
-                                _logger.LogError("[{buildFilePath}, {worker}, {layoutInd}: {srcPath} -> {dstPath}, {fileMetadata}] Destination path {dstPath2} already in use.", file.FullName, worker.GetType().FullName, i, srcPath, dstPath, fileMetadata, dstPath);
-                                return -1;
+                                if (!string.IsNullOrEmpty(fileMetadata.Sha512) &&
```

Comment:
This has the interesting side effect that if the first layout worker reports a file without a hash value and a subsequent worker reports the same file with a hash value, then this check will fail and publish will fail. We might want to consider hardening the hash assumptions to say that all reported files must have hash values, thus we don't have to think about whether they have hashes or not later.

I'm not asking for this to be corrected now, just something to think about that might simplify this in the future.

Reply:
This is mostly there as a safety check. All our implementations route `FileMetadata` creation through `Helpers.GetDefaultFileMetadata(file, FileClass.XXX)` which will always fill in the SHA512. 

I don't think that assumption is true for all the workers in the Diag repo, so we may need to harden their repo too.